tsdb/tsm1/cache_entry.go

```diff
@@ -117,6 +124,35 @@ func (e *entry) size() int {
 	return sz
 }
 
+// Contains returns true if values exist for the time interval [min, max]
+// inclusive. The entry is deduplicated prior to calling contains.
+func (e *entry) Contains(min, max int64) bool {
+	e.deduplicate()
```

Comment:
It's not clear to me that you need to de-duplicate values in the cache overlap the range.

Reply:
As previous, values are only appended to the slice, so if written out of order, this would be incorrect:

https://github.com/influxdata/influxdb/blob/a2a90ce87189c0738bddb5c9f27d71d59e060b8a/tsdb/tsm1/cache_entry.go#L76-L77

I can do the same as what I propose for `MaxTime`; scan for the min and max values in a single pass and avoid the deduplicate call.

Reply:
Yep, makes sense.